workflow-ui/backstack-android/src/main/java/com/squareup/workflow1/ui/backstack/StateRegistryHolder.kt

```diff
@@ -0,0 +1,222 @@
+package com.squareup.workflow1.ui.backstack
+
+import android.os.Bundle
+import androidx.lifecycle.Lifecycle.Event
+import androidx.lifecycle.Lifecycle.Event.ON_CREATE
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.savedstate.SavedStateRegistryController
+import androidx.savedstate.SavedStateRegistryOwner
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+
+/**
+ * Manages a group of [SavedStateRegistryOwner]s that are all saved to and restored from a single
+ * "parent" [SavedStateRegistryOwner].
+ *
+ * To save a child registry to the parent registry, call [saveRegistryController]. This should be
+ * done whenever the "navigation frame" associated with the registry is about to be hidden, and from
+ * the [onWillSave] callback. To restore a child registry, call [restoreRegistryControllerIfReady].
+ * This should be done when a navigation frame that was previously hidden is being shown again, and
+ * from the [onRestored] callback. See the kdoc on these methods and callbacks for more information.
+ *
+ * [attachToParentRegistry] should be called when the owning view is attached to a window, and
```

Comment:
There is no such thing as `ViewSavedStateRegistryOwner`. `ViewTreeSavedStateRegistryOwner` is just a class that has two static methods, it can't be meaningfully instantiated.